service/controller/resource/tcnp/create.go

```diff
@@ -34,29 +34,32 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	// Ensure some preconditions are met so we have all neccessary information
 	// available to manage the TCNP CF stack.
 	{
-		if len(cc.Spec.TenantCluster.TCNP.AvailabilityZones) == 0 {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "availability zone information not yet available")
+		if cc.Status.TenantCluster.Encryption.Key == "" {
```

Comment:
The `tcnpencryption`  resource cancels now when it does not find the encryption key. So we need to check here and cancel as well in case it is not there.